src/core/pointcloud/qgspointclouddataprovider.cpp

```diff
@@ -291,6 +292,26 @@ struct MapIndexedPointCloudNode
         pointAttr[ QStringLiteral( "X" ) ] = x;
         pointAttr[ QStringLiteral( "Y" ) ] = y;
         pointAttr[ QStringLiteral( "Z" ) ] = z;
+
+        const QDateTime gpsBaseTime = QDateTime::fromSecsSinceEpoch( 315964809, Qt::UTC );
+        const int numberOfSecsInWeek = 3600 * 24 * 7;
+        const QgsCopcPointCloudIndex *recastedIndex = dynamic_cast<QgsCopcPointCloudIndex *>( mIndex );
+        if ( recastedIndex )
+        {
```

Comment:
I would suggest to add a comment what's going on here, and what are the two different meanings of the gps time (+the fact we are also handling the case where the flag is not set, but the number of seconds is high enough)